api/src/services/authentication.ts

```diff
@@ -428,6 +440,48 @@ export class AuthenticationService {
 		};
 	}
 
+	private async updateStatefulSession(
+		sessionRecord: Record<string, any>,
+		oldSessionId: string,
+		newSessionId: string,
+		sessionExpiration: Date,
+	) {
```

Comment:
I think the term `ID` in the session context is misleading as  `token` is the identifier (PK) of a session. 🤔 
If accepted, this suggestion needs to be propagated to comments etc.

```suggestion
	private async updateStatefulSession(
		sessionRecord: Record<string, any>,
		oldSessionToken: string,
		newSessionToken: string,
		sessionExpiration: Date,
	) {
```